tls/s2n_connection.c

```diff
@@ -1290,3 +1292,27 @@ uint8_t s2n_connection_get_protocol_version(const struct s2n_connection *conn)
     }
     return conn->server_protocol_version;
 }
+
+int s2n_connection_set_keyshare_by_group_for_testing(struct s2n_connection *conn, uint16_t iana_id)
+{
+    S2N_ERROR_IF(!S2N_IN_TEST, S2N_ERR_NOT_IN_TEST);
+    notnull_check(conn);
+
+    if (iana_id == 0) {
+        conn->preferred_key_shares |= 1;
+        return 0;
+    }
+
+    const struct s2n_ecc_preferences *ecc_pref = NULL;
+    GUARD(s2n_connection_get_ecc_preferences(conn, &ecc_pref));
+    notnull_check(ecc_pref);
+
+    for (size_t i = 0; i < ecc_pref->count; i++) {
+                if (ecc_pref->ecc_curves[i]->iana_id == iana_id) {
+            conn->preferred_key_shares |= ( 1 << ( i + 1 ));
```

Comment:
I think a macro/function helper here would be super useful for readers who forget how this variable is encoded.